template/cmd/{{Executable}}/main.go

```diff
@@ -18,7 +17,8 @@ import (
 )
 
 var (
-	version = "unknown"
+	version  = "unknown"
+	settings = plugin.Settings{}
```

Comment:
It should be enough to define this within the main function instead of defining a global variable.

Reply:
I thought ideally we wanted to use Destination within urfave like this over in the `config.go` file.

```
func settingsFlags() []cli.Flag {
	return []cli.Flag{
		&cli.StringFlag{
			Name:    "github-url",
			Usage:   "github url, defaults to current scm",
			EnvVars: []string{"PLUGIN_GITHUB_URL", "DRONE_REPO_LINK"},
			Destination: &settings.GitHubURL,
		},
```

Reply:
I generally dislike global variables, especially if they don't need to be set from outside like the version. It should be supplied by parameter. Than you could even write some tests which could easily run in parallel, this is not possible with global variables.

Reply:
I modified the function to take a pointer to `plugin.Settings` so it can be populated by urfave. I don't think we can technically get around having a global in `main.go` because it needs to be present in `main` and `run` but if there's some go pattern I'm not familiar with let me know and I'll update with.